app/moderation/ModerationPageContent.tsx

```diff
@@ -0,0 +1,1437 @@
+import CommentsNode from '@/components/comments/CommentsNode';
+import Loading from '@/components/vulcan-core/Loading';
+import { useQuery } from '@/lib/crud/useQuery';
+import { gql } from '@/lib/generated/gql-codegen';
+import classNames from 'classnames';
+
+const ModeratorCommentsMultiQuery = gql(`
+  query multiModeratorCommentsQuery($commentIds: [String!]!) {
+    comments(input: { terms: { view: "commentsByIds", commentIds: $commentIds } }) {
+      results {
+        ...ShortformComments
+      }
+    }
+  }
+`);
+
+// Type definitions
+export type ModerationUserMin = {
+  __typename?: 'User';
+  _id: string;
+  displayName: string | null;
+  slug: string;
+};
+
+export type ModerationPostMin = {
+  __typename?: 'Post';
+  _id: string;
+  title: string;
+  slug: string;
+};
+
+export type ModerationComment = {
+  _id: string;
+  userId: string | null;
+  postId: string | null;
+  deletedDate?: Date | null;
+  postedAt?: Date | null;
+  deletedReason?: string | null;
+  rejectedReason?: string | null;
+  deletedPublic?: boolean | null;
+  contents?: any;
+  user: ModerationUserMin | null;
+  deletedByUser?: ModerationUserMin | null;
+  reviewedByUser?: ModerationUserMin | null;
+  post: ModerationPostMin | null;
+};
+
+export type ModerationPost = {
+  _id: string;
+  title: string;
+  slug: string;
+  userId: string | null;
+  createdAt?: Date;
+  postedAt?: Date | null;
+  rejectedReason?: string | null;
+  contents?: any;
+  user: ModerationUserMin | null;
+  reviewedByUser?: ModerationUserMin | null;
+  bannedUsers?: ModerationUserMin[];
+};
+
+export type ModerationUser = {
+  _id: string;
+  displayName: string | null;
+  slug: string;
+  bannedFrontpageUsers: ModerationUserMin[];
+  bannedPersonalUsers: ModerationUserMin[];
+};
+
+export type ModeratorComment = {
+  __typename: 'Comment';
+  _id: string;
+  userId: string | null;
+  postId: string | null;
+  postedAt?: Date | null;
+  contents?: any;
+  user: ModerationUserMin | null;
+  post: ModerationPostMin | null;
+};
+
+export type ActiveRateLimitDetails = {
+  actionType: string;
+  rateLimitType: string;
+  rateLimitCategory: "static" | "rolling" | "timed";
+  itemsPerTimeframe: number;
+  timeframeLength: number;
+  timeframeUnit: string;
+  rateLimitMessage: string;
+  activatedAt: Date;
+};
+
+export type ModerationUserFull = {
+  _id: string;
+  displayName: string | null;
+  slug: string;
+  createdAt?: Date | null;
+  karma?: number | null;
+  postCount?: number | null;
+  commentCount?: number | null;
+};
+
+export type ActiveRateLimit = {
+  userId: string;
+  rateLimits: ActiveRateLimitDetails[];
+  mostRecentActivation: Date;
+  user: ModerationUserFull | null;
+};
+
+export type GloballyBannedUser = {
+  _id: string;
+  displayName: string | null;
+  slug: string;
+  karma: number | null;
+  createdAt: Date | null;
+  postCount: number | null;
+  commentCount: number | null;
+  banned: Date | null;
+};
```

Comment:
I think maybe literally all of these types should've been auto-generated already?

Reply:
I don't think so. It's not like we are going through any graphql layer with the fragments

Reply:
Oh, yeah, if these are just the transformed db rows, then nvm.